src/locomoset/metrics/task_agnostic.py

```diff
@@ -0,0 +1,59 @@
+"""
+Functions for the task-agnostic metrics. In particular the number of parameters of a
+model as a metric.
+
+This also includes functions for reweighting a set of metric values by the inclusion of
+a task agnostic metric as a proxy for 'capactity to learn', as introduced in the
+Pairwise Annotation Representation Comparison (PARC) metric for transferability paper:
+
+Bolya, Daniel, Rohit Mittapalli, and Judy Hoffman. "Scalable diverse model selection
+for accessible transfer learning." Advances in Neural Information Processing Systems
+34 (2021): 19301-19312.
+"""
+
+import numpy as np
+from transformers import AutoModelForImageClassification
+
+
+def num_params_metric(model: AutoModelForImageClassification) -> int:
```

Comment:
I think we can be more generic in the type of model here, elsewhere we've used `from transformers.modeling_utils.PreTrainedModel` (also `AutoModelForImageClassification` is more like a wrapper for getting a model than a model itself).

Reply:
Ok, will refactor that.

Reply:
I've refactored this in [560fe72](https://github.com/alan-turing-institute/ARC-LoCoMoSeT/pull/47/commits/560fe7260eff453756ef47000d1db1797f1b1836) to use `transformers.modeling_utils.PreTrainedModel` as the type here and `transformers.AutoModel` for `rescale_by_params` for generality.